In [1]:
import pandas as pd
import numpy as np

In [3]:
data = pd.read_csv("Sachin_ODI.csv")
data.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [4]:
data.shape

(360, 14)

# Sample Space

In [5]:
len(data)

360

In [6]:
data.shape[0]

360

# Event A: Probability of India Winning given that Sachin has already scored a century

- P(India Win | Sachin Scored century)

In [12]:
pd.crosstab(index = data["century"], columns = data["Won"], margins = True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

# Observations:

- 160 is India losing and sachin not scoring a century
- 16 is India losing and Sachin scoring a century
- 176 is just India losing(Sample space for India losing)
- 154 India winning and Sachin not scoring a century
- 30 is India winning and Sachin scoring a century
- 184 India winning(Sample Space for India Winning)

- 314 is just Sachin not scoring a century(Sample space for Sachin not scoring century)
- 46 is Sachin Scoring a century(Sample Space for Sachin scoring century)

In [13]:
 30 / 46

0.6521739130434783

In [ ]:
# Probability of India Losing given that Sachin has scored a century

P(India Losing | Sachin scored a century)

In [ ]:
# Probability of Sachin scoring a century given that India has won the match

P(Sachin scoring century | India won the match)